src/Neo.VM/ExecutionEngine.cs

```diff
@@ -235,6 +235,13 @@ public ExecutionContext LoadScript(Script script, int rvcount = -1, int initialP
         protected virtual void OnFault(Exception ex)
         {
             State = VMState.FAULT;
+
+#if VMPERF
+            if (ex != null)
+            {
+                Console.Error.WriteLine(ex);
```

Comment:
Quite a useful one. Just for the record, in NeoGo we always log FAULT exceptions with WARN severity level:
```
2024-06-11T20:13:18.476+0300	WARN	contract invocation failed	{"tx": "3d82c9a5da51d2163895c67672380e5a4ba26cba1d400098d75ae82475e6a712", "block": 5782, "error": "at instruction 0 (PUSH1): gas limit is exceeded"}
```

So may be it's worth to consider add some more additional info to the message (like transaction hash, block number and etc). But it's just a suggestion.

Reply:
Currently this is only used for debugging purpose. But i get your point, we can have more logs like what you mentioned.